packages/core/src/api/clipboard/fromClipboard/pasteExtension.ts

```diff
@@ -26,6 +30,25 @@ function defaultPasteHandler({
   prioritizeMarkdownOverHTML: boolean;
   plainTextAsMarkdown: boolean;
 }) {
+  // Special case for code blocks, as they do not support any rich text
+  // formatting, so we force pasting plain text.
+  const selection = editor.prosemirrorView?.state.selection;
+  if (selection) {
+    const blockInfo = getBlockInfoFromSelection(editor.prosemirrorView.state);
+
+    const selectionInCodeBlock =
+      blockInfo.isBlockContainer && blockInfo.blockContent.node.type.spec.code;
+
+    if (selectionInCodeBlock) {
```

Comment:
Hm, can't we just check what the selection is pointing to, like:

```
tr.selection.$from.parent.type.spec.code&&
    tr.selection.$to.parent.type.spec.code
```

It makes it explicit that a selection may be pointing at different nodes, I'm unsure the right thing to do when only one of them is code. Whereas the getBlockInfoFromSelection kind of hides that.